netmiko/cloudgenix/cloudgenix_ion.py

```diff
@@ -0,0 +1,83 @@
+import time
+
+from netmiko import log
+from netmiko.cisco_base_connection import CiscoSSHConnection
+from netmiko.py23_compat import string_types
+
+
+class CloudGenixIonSSH(CiscoSSHConnection):
+    def establish_connection(self):
+        super(CloudGenixIonSSH, self).establish_connection(width=100, height=1000)
+
+    def session_preparation(self, *args, **kwargs):
+        """Prepare the session after the connection has been established."""
+        self.ansi_escape_codes = True
+        self._test_channel_read()
+        self.write_channel(self.RETURN)
+        self.set_base_prompt(delay_factor=5)
+
+    def disable_paging(self):
+        """Cloud Genix ION sets terminal height in establish_connection"""
+        return ""
+
+    def find_prompt(self, delay_factor=1):
+        prompt = super(CloudGenixIonSSH, self).find_prompt()
+        prompt = self.strip_backspaces(prompt).strip()
+        return prompt
+
+    def strip_command(self, command_string, output):
+        output = super(CloudGenixIonSSH, self).strip_command(command_string, output)
+        # handle command_string being repainted - otherwise return normal output
+        if len(output.split(command_string)) == 4:
+            output = output.split(command_string)[3]
+        return output
+
+    def config_mode(self):
+        """No configuration mode on ION SSH"""
+        self._in_config_mode = True
```

Comment:
@carlniger This is fine here, but I don't think this is the common pattern I have implemented for platforms that don't support config mode in Netmiko. In other words, I wouldn't generally want to propagate this Mikrotik patter for tracking config state forward.

Reply:
In general would we do something more like (for example) the PAN enable mode handling:

```
    def enable(self, *args, **kwargs):
        """No enable mode on PaloAlto."""
        pass
```

Just pass instead? And then pass again in `check_enable_mode`?